### ISSUE TITLE
vtkDICOMDirectory bad_alloc exception

### ISSUE THREAD

**Issue by VelazcoJD** [dgobbi/vtk-dicom#152]
Hello,

I am trying to use the vtkDICOMDirectory class to read DICOM data but I get bad_alloc exception. This occurs in line 2338 of vtkDICOMDirectory.cxx. Currently, I parse through the directory using boost library to get the list of files and load the data using the reader, but this is very limited. After reading the documentation, this class seems to offer what I need. Has any one encounter this issue before?

I am using Visual Studio 2015 (v140) tool set. I have the following code (like in the documentation).
```
vtkSmartPointer<vtkDICOMDirectory> dicomFinder = vtkSmartPointer<vtkDICOMDirectory>::New();
dicomFinder->SetDirectoryName(path.c_str());   // path is std::string
dicomFinder->RequirePixelDataOn();
dicomFinder->Update();	
```
The exception is thrown at Update function. Thanks in advance.

**Comment by dgobbi** [comment]
I haven't seen this error before.  As a test, can you download the win32 package from https://github.com/dgobbi/vtk-dicom/releases and run the enclosed "dicomfind.exe" on the directory?  It is a command-line program that uses vtkDICOMDirectory to search for files,  like so:

    dicomfind -image \path\to\directory

The "-image" option tells it to only search for image files, the full documentation is here: https://github.com/dgobbi/vtk-dicom/wiki/dicomfind

I'm curious about whether "dicomfind" will crash on that directory in the same way that vtkDICOMDirectory does.

**Comment by VelazcoJD** [comment]
Hello dgobbi,

Thanks for the quick response. I fixed the issue. I was able to execute the dicomfind.exe from the link you provided. Then, I checked the dicomfind.exe I built from source which also worked. Then I realized, I copied the debug dll instead of release dll! Everything else had worked so far that I didn't think to check that. Thanks for the help.

By the way, this is a great library you got here.
